corehq/apps/app_manager/models.py

```diff
@@ -3662,14 +3678,22 @@ def validate_app(self):
                 if xmlns_count[xmlns] > 1:
                     errors.append({'type': "duplicate xmlns", "xmlns": xmlns})
 
-        if self._has_parent_child_selection_cycle({m.unique_id:m for m in self.get_modules()}):
+        modules_dict = {m.unique_id: m for m in self.get_modules()}
+
+        def _parent_select_fn(module):
+            if hasattr(module, 'parent_select') and module.parent_select.active:
+                return module.parent_select.module_id
+
+        if self._has_parent_child_selection_cycle(modules_dict, _parent_select_fn):
             errors.append({'type': 'parent cycle'})
 
+        errors.extend(self._child_module_errors(modules_dict))
+
         if not errors:
             errors = super(Application, self).validate_app()
         return errors
 
-    def _has_parent_child_selection_cycle(self, modules):
+    def _has_parent_child_selection_cycle(self, modules, parent_id_fn):
```

Comment:
This used to detect a cycle in parent-child case list selection. Does it now also detect a cycle in module nesting? I think it's worth editing the docstring and/or name of the function to be clear about what this function does